ocpi-toolkit-2.2.1/src/main/kotlin/com/izivia/ocpi/toolkit/modules/sessions/SessionsCpoServer.kt

```diff
@@ -38,7 +38,7 @@ class SessionsCpoServer(
 
                 service
                     .getSessions(
-                        dateFrom = dateFrom?.let { Instant.parse(it) },
+                        dateFrom = Instant.parse(dateFrom),
```

Comment:
Instead of throwing an exception if the query parameter is not given and returning a server error (500), we should return an OCPI client error stating that there is a missing query parameter

From the docs:

> 2xxx: Client errors
> Errors detected by the server in the message sent by a client where the client did something wrong.

_https://github.com/ocpi/ocpi/blob/master/status_codes.asciidoc#2xxx-client-errors_

```suggestion
                        dateFrom = dateFrom?.let { Instant.parse(it) }
                            ?: throw OcpiClientInvalidParametersException("missing required date_from query parameter"),
```

Reply:
Absolutely agree.

Reply:
Fixed in c5cfc5458811dd6d3564e5a9a7ce09002df98586